cylc/flow/daemonize.py

```diff
@@ -70,22 +70,24 @@ def daemonize(schd):
                     if (log_stat.st_mtime == old_log_mtime or
                             log_stat.st_size == 0):
                         continue
-                    for line in open(logfname):
-                        if schd.START_MESSAGE_PREFIX in line:
-                            workflow_url, workflow_pid = (
-                                item.rsplit("=", 1)[-1]
-                                for item in line.rsplit()[-2:])
-                        if schd.START_PUB_MESSAGE_PREFIX in line:
-                            pub_url = line.rsplit("=", 1)[-1].rstrip()
-                        if workflow_url and pub_url:
-                            break
-                        elif ' ERROR -' in line or ' CRITICAL -' in line:
-                            # ERROR and CRITICAL before workflow starts
-                            try:
-                                sys.stderr.write(open(logfname).read())
-                                sys.exit(1)
-                            except IOError:
-                                sys.exit("Workflow schd program exited")
+                    with open(logfname, 'r') as logfile:
```

Comment:
`with open()` is the better pattern.

Note that this:

```
with open() as h:
    h.write()
```

Is *not* equivalent to this:

```python
h = open()
h.write()
h.close()
```

As the `with open` brings extra error handling to the table.